### ISSUE TITLE
1.12.x breaks Authentication for Drone (oAuth?), Jenkins (Authentication Token)

### ISSUE THREAD

**Issue by twoequaldots** [go-gitea/gitea#12026]
- Gitea version (or commit ref):
1.12.0, 1.12.1

- Git version:
2.16.4

- Operating system:
Gentoo (Linux 4.14)

- Database (use `[x]`):
  - [x] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [ ] SQLite

- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [ ] Not relevant
  - [x] Not applicable

- Log gist:
https://gist.github.com/twoequaldots/913aab2df201da668f356ac6b6797262

- Additional Information:
- - Authentication Sources (LDAP) - Active Directory in particular is set up
- - Jenkins authentication uses Authentication Token of a service account that has it's authentication source set to "Local" (i.e. not using LDAP / AD)
- - Jenkins uses this [plugin](https://plugins.jenkins.io/gitea/)

## Description

CI/CD builds were running from services like Jenkins and Drone on 1.11.x.  We've started noticing problems with Jenkins on 1.12.x.  We then checked on Drone and noticed Drone would redirect to Gitea for authorization and then spin while trying to sync up repositories.  The errors sent back to drone were 401 (invalid credentials).

I think it might be related to how authentication is now handled.  For example, the logs show some things may be out of place like this line:
```
2020/06/22 21:14:56 ...dels/login_source.go:815:UserSignIn() [W] Failed to login 'Username for 'https' via 'ActiveDirectory': user does not exist [uid: 0, name: Username for 'https, keyid: 0]
```

It also is out of place here:
```
2020/06/22 21:14:56 ...m.io/xorm/core/db.go:277:ExecContext() [I] [SQL] UPDATE "access_token" SET "uid" = $1, "name" = $2, "token_hash" = $3, "token_salt" = $4, "token_last_eight" = $5, "updated_unix" = $6 WHERE "id"=$7 [385 jenkins_dev 4d7f3ed9193e2aa48fcaa00c14511f45edb599542f0bbc5c190b37e42758d90baa665a89ffd12cb3fef39887d89affe9bb85 TeAlMLoClW 2125189f 1592860496 7] - 2.965272ms
```

Not entirely sure where it got `jenkins_dev` from but that isn't the service account.  The actual service account username is `svcjenkins` (the id is right though, 385).

We thought it might be related to this #11015 but maybe not so sure.  It might be related to this [commit](https://github.com/go-gitea/gitea/commit/1645d4a5d8def3cc5451e068aa0a321e028a889b#diff-455da54e97ad39fc570175e880682f83).

It's also possible that this may be Jenkins plugin problem but not entirely sure why Drone would stop working as well.

Let me know any other information that might be useful in debugging.  Thanks for all the hard work in advance :)

**Comment by dballard** [comment]
Going into my gitea user setting and deleting all old tokens under 'applications' suddenly allowed me to log into drone with gitea as auth again

**Comment by twoequaldots** [comment]
So after some digging, I was able to find the problem with Gitea <-> DroneCI.  This [PR](https://github.com/drone/go-scm/pull/66) hints at what is happening but basically it is a pagination error between DroneCI <-> Gitea.  The workaround I used is to increase the pagination settings in Gitea [conf](https://docs.gitea.io/en-us/config-cheat-sheet/#api-api).

As far as Jenkins, I changed the Jira plugin to use Basic Auth instead of an application token.  I'm going to try to delete all applications tokens for our Gitea user that Jenkins uses and then try with a new token to see if that fixes it (I already tried with a new token but didn't delete the old one).
